content/controllers.md

```diff
@@ -132,7 +132,7 @@ The request object represents the HTTP request and has properties for the reques
       <td><code>req.query</code> / <code>req.query[key]</code></td>
     </tr>
     <tr>
-      <td><code>@Headers(name?: string)</code></td>
+      <td><code>@Header(name?: string)</code></td>
```

Comment:
`@Header()` is a method decorator, for telling Nest to add a specific header to the response. `@Headers()` is a parameter decorator, to tell  Nest to retrieve `req.header` or `req.headers[name]`

Reply:
Thank you for your detailed and quick explanation. I confused them. But maybe they are easy to confuse. Since `@Headers()` is not consistent with `@Query()`, `@Param()`, the latters are singular, but the former is plural. Am I right?

Reply:
Well, `req.query` is what `@Query()` relates to. `req.params` is `@Param()`, but there's no setting of url parameters like there is of response headers. Express and Fastify's API for the `response` object use `header` for setting a header and for the request object use `headers` for retrieving the headers of the incoming request, so we're honestly just following the example they have.

Reply:
I consult the Fastify API, the `request` object did use `headers` to retrieve headers from the incoming request, but it also use `params` to retrieve the path parameters. They keep the same case, so they are consistent in Fastify.

refs:
- [Fastify Request API](https://www.fastify.io/docs/latest/Request/)
- [Fastify Reply API](https://www.fastify.io/docs/latest/Reply/)

Reply:
That's what I said, isn't it? `header` for setting, `headers` for getting.

`@Param()` maps to `req.params`, which is different, yes, but there's also no way to set URL parameters on a response, so it's a moot point. `@Param()` and `@Params()` would technically be referencing the same thing.

Reply:
Oh, I got it. Thank you very much for your patience! 👍 I just started learning NestJS yesterday, and bought your official course, it was very helpful. NestJS is very awesome, I will prompt it to my friends in China. Thank you for your awesome work!